flutter/lib/main.dart

```diff
@@ -0,0 +1,58 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_todo/realm/init_realm.dart';
+import 'package:provider/provider.dart';
+import 'package:realm/realm.dart';
+import 'realm/app_services.dart';
+import 'realm/schemas.dart';
+import './screens/homepage.dart';
+import './screens/log_in.dart';
+
+const _APP_ID = 'todo-sync-msgyz';
+
+void main() => runApp(MultiProvider(providers: [
+      ChangeNotifierProvider<AppServices>(create: (_) => AppServices(_APP_ID)),
+      ProxyProvider<AppServices, Realm?>(
+        update: (context, app, previousRealm) {
+          if (app.currentUser != null) {
+            previousRealm?.close();
+            return initRealm(app.currentUser!);
+          }
+          return null;
+        },
+        dispose: (_, realm) => realm?.close(),
+      )
+    ], child: const App()));
```

Comment:
did this MultiProvider/ProxyProvider implementation to provide top-level access to both realm and auth, while having the behavior of realm conditional on auth.

needed to put realm top level so that `showModalBottomSheet()` has access to it ([see this issue](https://stackoverflow.com/questions/63818402/flutter-cant-access-provider-from-showmodalbottomsheet))